src/Streamly/Prelude.hs

```diff
@@ -333,6 +336,14 @@ head m =
         yield a _ = return (Just a)
     in (runStream (toStream m)) Nothing stop yield
 
+-- | Extract all but the first element of the stream, if any.
+tail :: (Streaming t, Monad m) => t m a -> m (Maybe (t m a))
```

Comment:
The return type should be `Maybe (t m a)`.

Reply:
Actually I don't think this is possible consider the following example. I only assumed that I can create a function `tail` with the given return type and thereby created a function of type `String -> Int` which gives the number of bytes in a file.
```haskell
import Data.Maybe
import Streamly
import Streamly.Prelude ((.:),nil)

-- tail (nil) ~ Nothing
-- tail (pure 3) ~ Just nil
-- tail (pure 3 <> m) ~ Just m
sTail :: Streaming t => t m a -> Maybe (t m a)
sTail = undefined

-- Therefore I can easily define null as follows
-- which already escapes the monadic context
sNull :: Streaming t => t m a -> Bool
sNull m = isNothing $ sTail m

-- And when I have this I can do interesting things.
-- e.g. defining length without a monadic result.
sLength :: Streaming t => t m a -> Int
sLength m = go 0 (Just m)
    where
        go _ Nothing = 0
        go acc (Just m)
            | sNull m = acc
            | otherwise = go (1+acc) (sTail m)
          
-- Now I can easily escape arbitrary data from its `IO` context.
-- Lets just get the number of bytes in a File
numBytes :: String -> Int
numBytes fName = sLength . serially $ byteStream
    where 
      byteStream :: StreamT IO Char
      byteStream = do
        bytes <- liftIO $ readFile fName
        let toStreaming = foldr (.:) nil
        toStreaming bytes

main = print "tail therefore can't be t m a -> Maybe (t m a)"
```

Reply:
I am sorry to have said that, it is obviously incorrect to have the type as `Maybe (t m a)`.